src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -350,6 +370,35 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 }
 
 
+/*
+ * FindRelationShard finds the RelationShard for shard relation with
```

Comment:
I guess the given table is the oid of shell table (the empty one sitting on the coordinator). So we can change the comment as:

```
FindRelationShard finds the RelationShard for the shell table with the given
Oid  if exists in given relationShardList. Otherwise, returns NULL
```

If that is the case, we can even rename the parameter as `shellRelationId`. @onderkalaci could you quickly verify if this is the case indeed?

Reply:
Although the function is currently called for the situation you mentioned, this function can be more generic. It gets an inputRelationId, and a list of `RelationShard` to find if there is a matching entry.

I guess there is not much need for explicitly naming the function/params for the callers?

Though, if you feel strong about this, sure we can do.